forge-gui/res/cardsfolder/upcoming/riveteers_provocateur.txt

```diff
@@ -0,0 +1,12 @@
+Name:Riveteers Provocateur
+ManaCost:B R
+Types:Creature Viashino Warrior
+PT:3/1
+K:Menace
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChoose | RememberChosen$ True | TriggerDescription$ Whenever CARDNAME enters the battlefield, choose a creature card in your hand without blitz. It perpetually gains blitz. The blitz cost is equal to its mana cost.
```

Comment:
You really don't need the RememberChosen

Remembered and ChosenCard are different lists a Card can have

It's just that ChooseCardEffect automatically sets ChosenCard (because that's its main purpose anyway)